### PR TITLE
add support for local ssd recovery timeout field to instance and instance template.

### PR DESCRIPTION
**Pull request by muhammedkamal97**
add support for local ssd recovery timeout field to instance and instance template.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `local_ssd_recovery_timeout` field to `google_compute_instance` resource (beta)
compute: added `local_ssd_recovery_timeout` field to `google_compute_instance_template` resource (beta)
compute: added `local_ssd_recovery_timeout` field to `google_compute_regional_instance_template` resource (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/go.mod.erb
Comment (trodge) [inline mmv1/third_party/terraform/go.mod.erb:1]: Are all the changes here intentional?

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/go.mod.erb
Comment (muhammedkamal97) [reply mmv1/third_party/terraform/go.mod.erb:1]: yes updating the golang/api version since v132 is the newest version that contain that new field

Comment (trodge) [reply mmv1/third_party/terraform/go.mod.erb:1]: Do all of these packages need to be updated as well?

**Comment by trodge** [comment]
It looks like this change would also add these fields to the `google_compute_instance_from_machine_image` and `google_compute_instance_from_template` resources. Would it be possible to add tests for the fields in these resources as well?

**Review by trodge — APPROVED** [review]

**Pull request merged**